### PR TITLE
Docstring

### PR DESCRIPTION
**Pull request by AHReccese**
#### What does this implement/fix? Explain your changes.
Docstrings for the pymilo core package(not including tests) added.

### CODE REVIEW

File: pymilo/chains/linear_model_chain.py [hunk 1]
```diff
@@ -21,17 +21,41 @@
 
 
 def is_linear_model(model):
+    """
+    Returns whether the associated model is a sklearn's linear model or not
```
Comment (sadrasabouri) [inline pymilo/chains/linear_model_chain.py:25]: Change it to
```
Return whether the associated model is a sklearn's linear model or not
```
And any other places it occurs.

**Review by sadrasabouri — CHANGES_REQUESTED** [review]

**Comment by sadrasabouri** [comment]
@AHReccese I think we need to install `pydocstyle` in `test.yml`.
We've done it in PyCM by adding it to `dev-requirements` like [this](https://github.com/sepandhaghighi/pycm/blob/master/dev-requirements.txt#L8).

File: pymilo/transporters/baseloss_transporter.py [hunk 2]
```diff
@@ -145,6 +154,14 @@ def serialize(self, data, key, model_type):
             return data[key]
 
     def get_deserialized_base_loss(self, model_type, content):
+        """
+        extract the original BaseLoss object out of the associated core data recorded by pymilo.
+        :param model_type: the model type of the ML model, which it's data dictionary is given as the data param.
```
Comment (sadrasabouri) [inline pymilo/transporters/baseloss_transporter.py]: An extra blank line is needed between these two lines.
```
        extract the original BaseLoss object out of the associated core data recorded by pymilo.

        :param model_type: the model type of the ML model, which it's data dictionary is given as the data param.
```

File: .pydocstyle [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# my_pydocstyle_config.ini
```
Comment (sadrasabouri) [inline .pydocstyle:1]: remove this line.

**Review by sadrasabouri — CHANGES_REQUESTED** [review]

**Review by sadrasabouri — APPROVED** [review]
LGTM :rocket:

**Comment by AHReccese** [comment]
@alirezazolanvari 
It's ready for your semantic review.
Thanks.

File: pymilo/exceptions/deserialize_exception.py [hunk 4]
```diff
@@ -1,15 +1,42 @@
...
+    """An enum class to determine the type of deserialization errors."""
+    
     CORRUPTED_JSON_FILE = 1
     INVALID_MODEL = 2
     VALID_MODEL_INVALID_INTERNAL_STRUCTURE = 3
 
 
 class PymiloDeserializationException(PymiloException):
+    """
+    PymiloDeserializationException handle exceptions associated with Deserializations.
```
Comment (sadrasabouri) [inline pymilo/exceptions/deserialize_exception.py:17]: Just for the consistency, we may need to change it to:
```
Handle exceptions associated with Deserializations.
```

File: pymilo/exceptions/serialize_exception.py [hunk 5]
```diff
@@ -1,14 +1,37 @@
...
 class SerilaizatoinErrorTypes(Enum):
+    """An enum class used to determine the type of serialization errors."""
+    
     INVALID_MODEL = 1
     VALID_MODEL_INVALID_INTERNAL_STRUCTURE = 2
 
 
 class PymiloSerializationException(PymiloException):
+    """
+    PymiloserializationException handles exceptions associated with Serializations.
```
Comment (sadrasabouri) [inline pymilo/exceptions/serialize_exception.py:17]: Also here:
```
Handle exceptions associated with Serializations.
```

**Review by sadrasabouri — COMMENTED** [review]
Great job folks.
@alirezazolanvari please take a look at my comments and fix them if they are real issues.

File: pymilo/exceptions/deserialize_exception.py (hunk 4 above)
Comment (AHReccese) [reply pymilo/exceptions/deserialize_exception.py]: applied.

File: pymilo/exceptions/serialize_exception.py (hunk 5 above)
Comment (AHReccese) [reply pymilo/exceptions/serialize_exception.py]: applied.

**Pull request merged**